doc/application/index.rst

```diff
@@ -641,6 +641,73 @@ Zephyr binary into your application directory.
 You can also define the ``SOC_ROOT`` variable in the application
 :file:`CMakeLists.txt` file.
 
+.. _ext-projs:
+
+Modules (External projects)
+***************************
+
+Zephyr relies on the source code of several externally maintained projects in
+order to avoid reinventing the wheel and to reuse as much well-established,
+mature code as possible when it makes sense. In the context of Zephyr's build
+system those are called *modules*.
+There are several categories of external projects that Zephyr depends on,
+among which:
+
+- Debugger integration
+- Silicon Vendor Hardware Abstraction Layers (HALs)
+- Cryptography libraries
+- Filesystems
+- Inter-Process Communication (IPC)
+
+These modules (external projects) must be integrated with the Zephyr
+build system, which is based around Kconfig and CMake (see
+:ref:`application` for more information about Zephyr's build system).
+
+The way that the build system discovers and includes external projects when
+building an application is based around the :ref:`west` tool's ``west list``
+command. The main CMake script in zephyr (the :file:`CMakeLists.txt` located
+in the zephyr repository root folder) invokes ``west list`` to obtain a list of
+projects managed by west. For each project then the CMake scripts will verify
+if it contains the required metadata required in a module and, if it does,
+it will include it in the build.
```

Comment:
With three ``it``s in this sentence, I'm not sure what the last ``it`` is referring to here.  If it's a module, then say that:  ``... it will include that module in the build.``

Reply:
Tried to rephrase to make the sentence clearer.